specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2020-05-01-preview/machineLearningServices.json

```diff
@@ -4996,6 +5703,320 @@
+                  "type": "integer"
+                },
+                "InferenceDeploymentProvisioningState": {
+                  // https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property
+                  "type": "string",
+                  "enum": [
+                    "Unknown",
+                    "Scaling",
+                    "Creating",
+                    "Deleting",
+                    "Succeeded",
+                    "Failed",
+                    "Canceled"
+                  ],
+                  "x-ms-enum": {
+                    "name": "InferenceDeploymentProvisioningState",
+                    "modelAsString": true
+                  },
+                  "readOnly": true,
+                  "description": "The current provisioning state of the Inference Deployment resource."
+                },
+                "environmentConfigUri": {
+                  "type": "string",
+                  "description": "URI of the environment configuration Artifact."
+                },
+                "codeAssetUri": {
+                  "type": "string",
+                  "description": "URI of the Code Assets i.e. startup code."
+                },
+                "vnet": {
+                  "$ref": "#/definitions/VirtualNetworkProfile",
+                  "description": "Info of the customer's custom Virtual Network"
+                }
+              }
+            }
+          }
+        }
+      ]
+    },
+    "InferenceDeploymentCreateRequest": {
+      "description": "The request to create an Inference Deployment.",
+      "type": "object",
+      "properties": {
+        "models": {
+          "description": "The list of model URIs.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        // TODO: verify this is what autoscaler needs... also this seems to be in conflict
+        // with the sku in the base "Resource" class
+        "sku": {
+          "$ref": "#/definitions/SkuDescription"
+        },
+        "weight": {
+          "description": "weight of the deployment for load balancing.",
+          "type": "integer"
+        },
+        "InferenceDeploymentProvisioningState": {
+          // https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#provisioningstate-property
+          "type": "string",
+          "enum": [
+            "Unknown",
+            "Scaling",
+            "Creating",
+            "Deleting",
+            "Succeeded",
+            "Failed",
+            "Canceled"
+          ],
+          "x-ms-enum": {
+            "name": "InferenceDeploymentProvisioningState",
+            "modelAsString": true
+          },
+          "readOnly": true,
+          "description": "The current provisioning state of the Inference Deployment resource."
+        },
+        "environmentConfigUri": {
```

Comment:
what about environment variables and comandLine? are these properties of the environment? if yes does that mean we expect SDK to create a new environment every time the user wants to override a single env var?

Reply:
need to figure out what comes with environment config